### ISSUE TITLE
Provide contribution guide information for adding authentication sources

### ISSUE THREAD

**Issue by tomkerkhove** [kedacore/keda#2718]
Provide contribution guide information for adding authentication sources

**Comment by tomkerkhove** [comment]
@v-shenoy Since you recently did this, is this something you are interested in helping with?

**Comment by v-shenoy** [comment]
What kind of information do you expect, and where do I add this?

**Comment by tomkerkhove** [comment]
Something similar to https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#contributing-scalers that walks you through the process in case you are a new contributor for auth providers.

So basically what you had to figure out yourself, but now documented :D

**Comment by v-shenoy** [comment]
I am not sure how to add a documentation that will be generic enough to cover all authentication sources. So, as far as my understanding goes, to add an authentication source, you would need to make changes to the TriggerAuth/ClusterTriggerAuth CRD and have the user provide information to query the auth provider. 

Now using these values from the CRD, there need to be code changes to use these values.  These code changes can differ on the basis of auth source (ex - key vault vs workload identity), so I am a bit confused here.

**Comment by tomkerkhove** [comment]
@kedacore/keda-maintainers What do you think? Is this valuable? I might be overthinking it but want to make it simple for people to contribute more auth providers.

**Comment by zroubalik** [comment]
Yeah, I think that even a simple documentation is great! It doesn't have to be complex, but something that gives users some general overview.
